File: frontend/lib/rewardTag.ts
export interface RewardTag {
  reward_id: string;
  tag_id: string;
  user_id: string;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface RewardTagInput {
  reward_id: string;
  tag_id: string;
  deleted_at?: string | null;
}

/**
 * Creates a composite key for a reward-tag association
 */
export function rewardTagKey(rewardId: string, tagId: string): string {
  return `${rewardId}:${tagId}`;
}

/**
 * Parses a composite key back into reward_id and tag_id
 */
export function parseRewardTagKey(key: string): { rewardId: string; tagId: string } {
  const [rewardId, tagId] = key.split(":");
  return { rewardId, tagId };
}

File: ios/bochi/Rewards/Models/RewardTag.swift
import Foundation

// Junction model for the many-to-many Reward ↔ Tag relationship.
// Keeping reward links separate from recurringTask links means the global tag catalog
// stays shared while each feature owns its own association history.
struct RewardTag: Identifiable, Equatable, Sendable, Codable, OwnerScopedRecord {
    let rewardId: RecordID
    let tagId: RecordID
    let createdAt: Date
    let updatedAt: Date
    let deletedAt: Date?
    let serverRevision: Int64?

    var id: RecordID { RecordID("\(rewardId):\(tagId)") }

    init(
        rewardId: RecordID,
        tagId: RecordID,
        createdAt: Date,
        updatedAt: Date,
        deletedAt: Date?,
        serverRevision: Int64? = nil
    ) {
        self.rewardId = rewardId
        self.tagId = tagId
        self.createdAt = createdAt
        self.updatedAt = updatedAt
        self.deletedAt = deletedAt
        self.serverRevision = serverRevision
    }
}
